src/printer.js

```diff
@@ -3837,19 +3828,38 @@ function printJSXElement(path, options, print) {
     return openingLines;
   }
 
+  // Convert `{" "}` to text nodes containing a space.
+  // This makes it easy to turn them into `jsxWhitespace` which
+  // can then print as either a space or `{" "}` when breaking.
+  n.children = n.children.map(child => {
```

Comment:
I'm assuming that mutating the AST is acceptable. If that assumption isn't true let me know and I can re-work this to work on a copy of the children.